modules/generator/processor/servicegraphs/servicegraphs.go

```diff
@@ -177,14 +174,12 @@ func (p *processor) consume(resourceSpans []*v1_trace.ResourceSpans) error {
 					continue
 				}
 
-				// upsertEdge will only return this errTooManyItems
+				// UpsertEdge will only return ErrTooManyItems
 				if err != nil {
 					return err
 				}
 
-				if edge.IsCompleted() {
-					p.collectCh <- k
-				}
+				p.metricTotalEdges.Inc()
```

Comment:
Is this metric correct here? It looks like it should be called in `onComplete`.

Reply:
Okay, I've fixed this now by only incrementing it when a new edge is added. Returning `isNew` from `UpsertEdge` is a bit tedious but it keeps the Prometheus metrics out of the store implementation. We could alternatively also just move all metrics into store.